docs/decisions/infra/2025-01-09-notifications-architecture.md

```diff
@@ -0,0 +1,85 @@
+
+2. **Option 2 (Chosen)**: Use the domain of the environment (e.g., hosted zone).
+   - **Pros**:
+     - Leverages Amazon’s Easy DKIM, simplifying setup by automatically managing public/private keys.
+   - **Cons**:
+     - In multi-app repositories, applications that share environments (dev, staging, prod) and hosted zones cannot create their own domain email identity since it would create a naming conflict. Therefore, enabling notifications for a second app requires custom work to allow sharing of the notification resources from the first app.
+
+### Consideration 2: Where should notifications resources be defined?
+
+1. **Option 1**: Create resources as a new app/microservice shared by other apps.
+   - **Pros**:
+     - Allows for a single notifications sandbox environment to support multiple applications across multiple non-production environments
+     - Avoids duplication of resources across multiple apps
+     - Most modular, allowing for more flexible scaling and resource management.
+   - **Cons**:
+     - Adds operational complexity by requiring notifications to be separately deployed
+     - Requires notifications to be designed to support multiple tenants (multiple applications across multiple environments) which adds significant complexity within the notifications service.
+
+2. **Option 2**: Create resources in a new layer (e.g., build repository layer) shared across apps.
+   - **Cons**:
+     - Adds conceptual complexity to the infrastructure by creating a new separate layer
+     - Adds operational complexity by requiring the notifications layer to be separately deployed
+
+3. **Option 3**: Create resources in the service layer for each app needing notifications.
+   - **Pros**:
+     - Simplest to understand as it mirrors the structure of the rest of the infrastructure
+   - **Cons**:
+     - Only feasible if domains are not shared across multiple applications i.e. we use the domain of the app for sender email notifications. which was not the chosen option due to other factors.
+
+4. **Option 4 (Chosen)**: Create resources in the service layer of one app and require custom work for others to access the notifications service in the app that contains that service.
+   - **Pros**:
+     - As simple as Option 3 for projects where only one app needs to send notifications, which is the common case
+   - **Cons**:
+     - May require custom work in cases where more than one app needs to send notifications
+
+### Consideration 3: Which layer should contain DKIM and DMARC DNS Records?
+
+1. **Option 1**: Create records in the network layer.
```

Comment:
In retrospect I might have chosen this had I thought about it more. Particularly in the case of grants.gov where we don't have any control of what DNS records get published. The DNS being in a different module would make it deploy an application in a way that feels clearly demarcated

Reply:
I'm open to exploring this option, but would want to think about whether the pattern would also work for the A records for setting up the service's custom domain. The problem with the A records is you can only create those records after you know the DNS name for the load balancer which is in the service layer, and I'm hesitant (though open minded) to create a dependency on service layer resources in the network layer.